src/runtime/pythonengine.cs

```diff
@@ -64,7 +67,17 @@ public static string ProgramName
 
                 return result ?? "";
             }
-            set { Runtime.Py_SetProgramName(value); }
+            set
+            {
+                if (_programName != IntPtr.Zero)
```

Comment:
ok makes sense. My original implementation `dealloc` twice. A better check would have been to check against `null`. Ill fix it up